enterprise/frontend/src/embedding-sdk/hooks/private/get-auth-configuration.unit.spec.tsx

```diff
@@ -0,0 +1,75 @@
+import type { SDKConfig } from "embedding-sdk";
+import { getAuthConfiguration } from "embedding-sdk/hooks";
+import { getErrorMessage } from "embedding-sdk/lib/user-warnings/constants";
+import {
+  createMockApiKeyConfig,
+  createMockJwtConfig,
+} from "embedding-sdk/test/mocks/config";
+import api from "metabase/lib/api";
+
+const JWT_CONFIG = createMockJwtConfig();
+const API_KEY_CONFIG = createMockApiKeyConfig();
+const NO_CONFIG = {};
+
+const setup = ({ config }: { config: SDKConfig }) => {
+  const dispatch = jest.fn();
+  const appName = "Metabase";
+  return getAuthConfiguration(config, dispatch, appName);
+};
+
+jest.mock("metabase/lib/api");
+global.console = { ...global.console, warn: jest.fn() };
+
+describe("getAuthConfiguration", () => {
```

Comment:
added some tests here as I wanted to make sure the pattern matching works as intended